quinn/benches/bench.rs

```diff
@@ -12,46 +12,65 @@ use tracing_futures::Instrument as _;
 
 use quinn::{ClientConfigBuilder, Endpoint, ServerConfigBuilder};
 
-benchmark_group!(benches, large_streams, small_streams);
+benchmark_group!(
+    benches,
+    large_data_1_stream,
+    large_data_10_streams,
+    small_data_1_stream,
+    small_data_100_streams
+);
 benchmark_main!(benches);
 
-fn large_streams(bench: &mut Bencher) {
-    let _ = tracing_subscriber::fmt::try_init();
+const LARGE_DATA: &[u8] = &[0xAB; 1024 * 1024];
 
-    let ctx = Context::new();
-    let (addr, thread) = ctx.spawn_server();
-    let (endpoint, client, mut runtime) = ctx.make_client(addr);
-    const DATA: &[u8] = &[0xAB; 128 * 1024];
-    bench.bytes = DATA.len() as u64;
-    bench.iter(|| {
-        runtime.block_on(async {
-            let mut stream = client.open_uni().await.unwrap();
-            stream.write_all(DATA).await.unwrap();
-            stream.finish().await.unwrap();
-        });
-    });
-    drop(client);
-    runtime.block_on(endpoint.wait_idle());
-    thread.join().unwrap();
+const SMALL_DATA: &[u8] = &[0xAB; 1];
+
+fn large_data_1_stream(bench: &mut Bencher) {
+    send_data(bench, LARGE_DATA, 1);
 }
 
-fn small_streams(bench: &mut Bencher) {
+fn large_data_10_streams(bench: &mut Bencher) {
+    send_data(bench, LARGE_DATA, 10);
+}
+
+fn small_data_1_stream(bench: &mut Bencher) {
+    send_data(bench, SMALL_DATA, 1);
+}
+
+fn small_data_100_streams(bench: &mut Bencher) {
+    send_data(bench, SMALL_DATA, 100);
+}
+
+fn send_data(bench: &mut Bencher, data: &'static [u8], concurrent_streams: usize) {
     let _ = tracing_subscriber::fmt::try_init();
 
     let ctx = Context::new();
     let (addr, thread) = ctx.spawn_server();
     let (endpoint, client, mut runtime) = ctx.make_client(addr);
-    const DATA: &[u8] = &[0xAB; 1];
-    bench.bytes = 1;
+    let client = Arc::new(client);
+
+    bench.bytes = (data.len() as u64) * (concurrent_streams as u64);
     bench.iter(|| {
+        let mut handles = vec![];
```

Comment:
Why not use `FuturesUnordered` here? Also, if no arguments are passed we generally prefer `Vec::new()` over `vec![]` (on the premise that, all other things being equal, we prefer using fewer abstractions).

Reply:
I actually found just spawning all and then waiting in order to be the simpler abstraction. But there is not too much difference. It was just the first thing that came to my mind for this use-case.

It willl actually have an impact on performance if the non single-threaded tokio runtime is used:
- With `FuturesUnordered` everything will still run in on tokio task, and there will be no races on accessing the connection from there.
- With `tokio::spawn` tasks on multiple threads for access on the connection.

I think going for any of the 2 approaches as a baseline and then checking out differences in follow-ups is reasonable.
I can change this version to `FuturesUnordered` if you prefer.

Reply:
Leaving it as-is makes sense for reducing the changes needed in the future to perform a comparison between schedulers, I think. I don't expect the overhead reduction from `FuturesUnordered` to be huge here.